src/gen/lib/Pattern_name.mll

```diff
@@ -0,0 +1,65 @@
+{
+  (* Only pure lists of Char are guaranteed to be parsed correctly
+     by our parser. *)
+  type element =
+    | Char of char (* lowercased ascii *)
+    | Other (* something that we couldn't parse *)
+
+  let deduplicate xs =
+    let tbl = Hashtbl.create 10 in
+    List.filter (fun x ->
+      if Hashtbl.mem tbl x then
+        false
+      else (
+        Hashtbl.add tbl x ();
+        true
+      )
+    ) xs
+
+  let normalize_char = function
+  | '-' -> '_'
+  | c -> Char.lowercase_ascii c
+
+  let lowercase_singleton_of_cset s =
```

Comment:
For consistency with `normalize_char`, and since we also modify `-`, maybe `normalized_singleton_of_cset`?

Reply:
thanks!